### PR TITLE
JSCS validation changes

### PR DESCRIPTION
**Pull request by samdutton**
Updates samples to validate with JSCS using Google rules.

I've kept samples/web/.jscsrc  with `"preset": "google"` (rather than just adding `preset: "google"` to Gruntfile.js) because that causes SublimeLinter-jscs to use Google rules by default (since it looks for .jscsrc).

@andresusanopinto @juberti @jiayliu @chuckhays  PTAL

### CODE REVIEW

**Comment by samdutton** [comment]
See #402.

File: Gruntfile.js [hunk 1]
```diff
@@ -47,17 +47,13 @@ module.exports = function(grunt) {
...
         'samples/web/content/apprtc/js/compiled/*.js',
         'samples/web/content/apprtc/js/vr.js',
         'samples/web/content/apprtc/js/stereoscopic.js',
         'samples/web/content/getusermedia/desktopcapture/extension/content-script.js',
-        'samples/web/content/testrtc/bower_components/**',
-        // TODO (chuckhays) : remove these exclusions after code passes.
-        'samples/web/content/getusermedia/**/*.js',
-        'samples/web/content/peerconnection/**/*.js',
-        'samples/web/content/datachannel/**/*.js'
+        'samples/web/content/testrtc/bower_components/**'
```
Comment (andresusanopinto) [inline Gruntfile.js]: You can remove this line too.

File: Gruntfile.js [hunk 2]
```diff
@@ -47,17 +47,13 @@ module.exports = function(grunt) {
     jscs: {
       src: 'samples/web/content/**/*.js',
       options: {
-        preset: 'google', // as per Google style guide – could use '.jscsrc' instead
+        config: 'samples/web/.jscsrc', // uses Google preset with some exceptions
```
Comment (andresusanopinto) [inline Gruntfile.js]: I think overall we should aim at the google preset with no modifications/exceptions.

Is fixing the 100chars lines and the multipleVarDecl too much to go in this CL?
Or you plan to never fix them?
If the first, then add a TODO and follow up with a CL to clean it up.

File: Gruntfile.js (hunk 1 above)
Comment (samdutton) [inline Gruntfile.js]: Good spot – removed these references to testrtc components in next commit.

**Comment by juberti** [comment]
please rebase on new structure

**Comment by samdutton** [comment]
Are we happy with the /samples structure proposed in the doc
<https://docs.google.com/document/d/1Wt7Zwa1tJ3glpiY-7yQ_retMMVnCJECD5pzbiwKW7zo/edit#heading=h.tfyv6tshunvu>?
If so, I'll rearrange it like that tomorrow (Friday).

**Comment by juberti** [comment]
I think it is in the correct form now.

**Comment by samdutton** [comment]
There are a couple of differences between the version in the doc and the
version in the repo:
* there's no build directory in the repo at present: /src has .csslintrc
and .jshhintrc
* /src has a content directory that contains the samples rather than
directories for datachannel, getusermedia and peerconnection.

I don't mind not having a build directory, but I'd prefer to remove the
content directory layer.

One option could be to have a very shallow structure, which also makes the
Github Pages URLs nice and short:

.csslintrc
.gitignore
.jshintrc
.travis.yml
CONTRIBUTING.md
index.html
LICENSE.md
README.md
Gruntfile.js
css/
datachannel/
getusermedia/
js/
peerconnection/

**Comment by samdutton** [comment]
I've updated to fit the new structure – JSCS and JSHint now lint all files with no exclusions.

@andresusanopinto

File: src/content/peerconnection/munge-sdp/js/main.js [hunk 3]
```diff
@@ -9,7 +9,8 @@
 'use strict';
 
 var getMediaButton = document.querySelector('button#getMedia');
-var createPeerConnectionButton = document.querySelector('button#createPeerConnection');
+var createPeerConnectionButton =
+  document.querySelector('button#createPeerConnection');
```
Comment (juberti) [inline src/content/peerconnection/munge-sdp/js/main.js]: indent 4 here

File: src/content/peerconnection/trickle-ice/js/main.js [hunk 4]
```diff
@@ -42,7 +43,8 @@ function addServer() {
 
   // Store the ICE server as a stringified JSON object in option.value.
   var option = document.createElement('option');
-  var iceServer = createIceServer(urlInput.value, usernameInput.value, passwordInput.value);
+  var iceServer = createIceServer(urlInput.value, usernameInput.value,
+    passwordInput.value);
```
Comment (juberti) [inline src/content/peerconnection/trickle-ice/js/main.js]: indent 4

File: src/content/peerconnection/create-offer/js/main.js [hunk 5]
```diff
@@ -14,7 +14,8 @@ var vadInput = document.querySelector('input#vad');
 var videoInput = document.querySelector('input#video');
 
 var numAudioTracksInput = document.querySelector('div#numAudioTracks input');
-var numAudioTracksDisplay = document.querySelector('span#numAudioTracksDisplay');
+var numAudioTracksDisplay =
+  document.querySelector('span#numAudioTracksDisplay');
```
Comment (juberti) [inline src/content/peerconnection/create-offer/js/main.js]: indent 4

File: src/content/getusermedia/filter/js/main.js [hunk 6]
```diff
@@ -30,8 +30,8 @@ filterButton.onclick = function() {
   canvas.className = filters[newIndex];
 };
 
-navigator.getUserMedia = navigator.getUserMedia || navigator.webkitGetUserMedia ||
-  navigator.mozGetUserMedia;
+navigator.getUserMedia = navigator.getUserMedia ||
+  navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
```
Comment (juberti) [inline src/content/getusermedia/filter/js/main.js]: indent 4

File: src/content/getusermedia/gum/js/main.js [hunk 7]
```diff
@@ -15,8 +15,8 @@ var constraints = {
   video: true
 };
 
-navigator.getUserMedia = navigator.getUserMedia || navigator.webkitGetUserMedia ||
-  navigator.mozGetUserMedia;
+navigator.getUserMedia = navigator.getUserMedia ||
+  navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
```
Comment (juberti) [inline src/content/getusermedia/gum/js/main.js]: indent 4

File: src/content/datachannel/js/main.js [hunk 8]
```diff
@@ -67,13 +71,15 @@ function createConnection() {
     }
   }
   // Add localConnection to global scope to make it visible from the browser console.
-  window.localConnection = localConnection = new RTCPeerConnection(servers, pcConstraint);
+  window.localConnection = localConnection =
+    new RTCPeerConnection(servers, pcConstraint);
```
Comment (juberti) [inline src/content/datachannel/js/main.js]: all wrapped lines should indent 4. I mentioned several below, but there are many others.

**Comment by juberti** [comment]
/build seems like a good idea
regarding /src/content vs /src, I like the idea of having a place for code common to multiple samples. With adapter moving to being require-d, the need for this is less, but this still seems like a good idea. Right now the location for this is at src/js. So we could either keep the existing structure, or do something like

src/
  common
    css
    js
     videopipe.js
  gum
  dc
  pc
  third_party
    js
      face detect stuff

wdyt?

**Comment by samdutton** [comment]
With Sublime regexing, I've attempted to correct all the 4-space indents.

**Comment by samdutton** [comment]
Regarding structure, how about this:

.gitignore
.travis.hml
CONTRIBUTING.md
Gruntfile.js
LICENSE.md
README.md
index.html
package.json
src/

src/datachannel/
src/getusermedia/
src/global/ (or src/common/)
src/peerconnection/
.csslintrc (it might actually be neater to put these at the top level – that's what Web Fundamentals does)
.jscsrc
.jshintrc

src/datachannel/...
src/getusermedia/...
src/peerconnection/...

src/global/css/...
src/global/images/...
src/global/js/...

Or a shallower version, with the top level like this:

.csslintrc
.gitignore
.jscsrc
.jshintrc
.travis.hml
CONTRIBUTING.md
Gruntfile.js
index.html
LICENSE.md
package.json
README.md

css/
datachannel/
getusermedia/
images/
js/
peerconnection/

This is unusual in not having code in a src directory, but I like the simplicity and the GitHub Pages URLs that result. I think the 'globalness' of css/, js/ and images/ is implicit.

**Comment by samdutton** [comment]
Could we get a LGTM for the JSCS and JSHint changes?

Once those are incorporated, I can look at updating the repo structure.

**Comment by chuckhays** [comment]
The jscs and jshint changes lgtm.

**Comment by samdutton** [comment]
Includes fix to #477.

**Pull request merged**